jwst/resample/resample_spec.py

```diff
@@ -313,6 +315,91 @@ def _max_virtual_slit_extent(self, wcs_list, target_ra, target_dec):
 
         return y_slit_min, y_slit_max
 
+    def build_miri_lrs_output(self):
+        """ Build MIRI LRS FIXEDSLIT output WCS. """
+        all_wavelength = []
+        all_ra_slit = []
+        all_dec_slit = []
+
+        for im, model in enumerate(self.input_models):
+            wcs = model.meta.wcs
+            bbox = wcs.bounding_box
+            grid = wcstools.grid_from_bounding_box(bbox)
+            ra, dec, lam = np.array(wcs(*grid))
+            # Handle vertical (MIRI) or horizontal (NIRSpec) dispersion.  The
+            # following 2 variables are 0 or 1, i.e. zero-indexed in x,y WCS order
+            spectral_axis = find_dispersion_axis(model)
+            spatial_axis = spectral_axis ^ 1
+
+            # Compute the wavelength array, trimming NaNs from the ends
+            # In many cases, a whole slice is NaNs, so ignore those warnings
+            warnings.simplefilter("ignore")
+            wavelength_array = np.nanmedian(lam, axis=spectral_axis)
+            warnings.resetwarnings()
+            wavelength_array = wavelength_array[~np.isnan(wavelength_array)]
+
+            all_wavelength = np.append(all_wavelength, wavelength_array)
+            this_minw = np.min(wavelength_array)
+            this_maxw = np.max(wavelength_array)
+            all_minw = np.min(all_wavelength)
+            all_maxw = np.max(all_wavelength)
+            if this_minw < all_minw:
+                addpts = wavelength_array[wavelength_array < all_minw]
```

Comment:
I would add some `delta` to this inequality. For example,

```python
deltas = np.ediff1d(wavelength_array)
delta_min = deltas[:5].mean()
delta_max = deltas[-5:].mean()
if this_minw < all_minw - 0.3 * delta_min:
    addpts = wavelength_array[wavelength_array < (all_minw - 0.3 * delta_min)]
.......
if this_maxw > all_maxw + 0.3 * delta_max:
    addpts = wavelength_array[wavelength_array > (all_maxw + 0.3 * delta_max)]
```

This helps avoid situations in which `this_minw` is barely smaller than `all_minw`. I have encountered this in the past, see https://github.com/spacetelescope/jwst/pull/7321